pkg/core/id.go

```diff
@@ -42,6 +42,27 @@ func (i ID) String() string {
 	return fmt.Sprintf("%s, %s/%s", i.GroupKind.String(), i.Namespace, i.Name)
 }
 
+// FromID returns a GKNN representing the provided ID string.
+func FromID(id string) (ID, error) {
```

Comment:
core.FromID is a confusing name, since it doesn't return a "core". How about ParseID?

The func comment should also explain the format.

Reply:
The comment also says it returns a GKNN, but it doesn't... it returns an ID.